waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -61,6 +62,16 @@ type
     serviceSlots*: Table[string, RemotePeerInfo]
     started: bool
 
+proc protocolMatcher*(codec: string): Matcher =
+  ## Returns a protocol matcher function for the provided codec
+  proc match(proto: string): bool {.gcsafe.} =
+    ## Matches a proto with any postfix to the provided codec.
+    ## E.g. if the codec is `/vac/waku/filter/2.0.0` it matches the protos:
+    ## `/vac/waku/filter/2.0.0`, `/vac/waku/filter/2.0.0-beta3`, `/vac/waku/filter/2.0.0-actualnonsense`
+    return proto.startsWith(codec)
+
+  return match
```

Comment:
mm good catch, will remove it

Reply:
wops, actually used by the waku_node in `startRelay`. adding to the todo list since reconnection logic should be in peermanager side, and waku_node should just call a high level function.